core/src/main/java/org/apache/calcite/sql/validate/DelegatingScope.java

```diff
@@ -298,7 +298,11 @@ public SqlQualified fullyQualify(SqlIdentifier identifier) {
       for (; i > 0; i--) {
         final SqlIdentifier prefix = identifier.getComponent(0, i);
         resolved.clear();
-        resolve(prefix.names, nameMatcher, false, resolved);
+        resolve(
+            prefix.names,
+            nameMatcher,
+            validator.getConformance().allowAliasUnnestArrayColumns(),
+            resolved);
```

Comment:
The code may works for your case, but you have changed a common sql identifier resolving logic with a special conformance, which seems not right way to go ~

The conformance switch logic should always happen in SqlValidator, we can pass along some switch flags into the scope if necessary.

Reply:
Sorry a little confused here. Here I am directly using conformance as the flag. I feel like it's more concise than an explicit flag like 

SqlValidatorImpl.java::
boolean allowAliasUnnestArrays = validator.getConformance().allowAliasUnnestArrayColumns;
DelegatingScope.setAllowAliasUnnestArrays(allowAliasUnnestArrays);

DeletatingScope.java::
Just replace validator.getConformance().allowAliasUnnestArrayColumns with the explicit flag.

And it seems using conformance in scope happens before. e.g. OrderByScope.

Reply:
The current fix seems not right, the nested field can be seen if we set up the record type to `StructKind.PEEK_FIELDS`, for the test cases you gave:
```java
sql("select e.ENAME\n"
        + "from dept_nested as d CROSS JOIN\n"
        + " UNNEST(d.employees) as t(e)")
      .withConformance(SqlConformanceEnum.PRESTO).columnType("VARCHAR(10) NOT NULL");
```
There `t(e)` table has record type with `e` as field, the `e` can be seen and if you change the nested employee type with  code snippet:
```java
final RelDataType empRecordType = typeFactory.builder()
      .add("EMPNO", intType)
      .add("ENAME", varchar10Type)
      .add("DETAIL", typeFactory.builder()
          .add("SKILLS", array(skillRecordType)).build())
      .kind(StructKind.PEEK_FIELDS) // here is the line you can tweak
      .build();
```

So, in total, this line change is not necessary, the field access should not be controlled by the `SqlConformance`.

Reply:
make sense. Will do